azure/durable_functions/models/DurableOrchestrationContext.py

```diff
@@ -158,19 +161,38 @@ def call_activity(self, name: str, input_: Optional[Any] = None) -> TaskBase:
         Task
             A Durable Task that completes when the called activity function completes or fails.
         """
+        if(isinstance(name, Callable)):
+            if (isinstance(name, FunctionBuilder)):
+                if (isinstance(name._function._trigger, ActivityTrigger)):
+                    name = name._function._name
+                else:
+                    error_message = "Received function with Trigger-type `"\
+                        + name._function._trigger.type\
+                        + "` but expected `ActivityTrigger`."\
+                        "Ensure your function is annotated with the `activity_trigger` decorator"\
+                        "or directly pass in the name of the activity as a string."
+                    raise Exception(error_message)
+            else:
+                error_message = "Received a callable function without an associated trigger-type."\
+                    "Please ensure you're using the Python programming model V2 "\
+                    "and that your function is annotated with the `activity_trigger` decorator."\
+                    "Otherwise, directly pass in the name of the activity as a string."
+                raise Exception(error_message)
+
```

Comment:
This logic seems to be shared with the `call_activity_with_retry` method. Mind factoring it out into a helper function?

Reply:
Relatedly, do you mind adding some conditional checks to ensure that the variable `name` actually has the properties `_function` and `_trigger` and `_name`? You can do this via `has_attr`, which is described here: https://stackoverflow.com/questions/610883/how-do-i-check-if-an-object-has-an-attribute/610923#610923

I think this would be good as we're technically accessing "internal variables" in order to validate this logic, so it would be wise to add checks in case these variables do not exist.